docs/standard/native-interop/qualify-net-types-for-interoperation.md

```diff
@@ -40,10 +42,237 @@ If you intend to expose types in an assembly to COM applications, consider the r
+| 0 | `IUnknown::QueryInterface` |
+| 1 | `IUnknown::AddRef` |
+| 2 | `IUnknown::Release` |
+| 3 | `IComInterface::Method` |
+| 4 | `IComInterface::Method2` |
+| 5 | `IComInterface2::Method3` |
+
+As a result, it is very easy to call a method defined on `IComInterface` from an `IComInterface2*`. Specifically, calling a method on a base interface does not require a call to `QueryInterface` to get a pointer to the base interface. Additionally, C++ allows an implicit conversion from `IComInterface2*` to `IComInterface*`, which is well defined and allows avoiding a `QueryInterface` call again. As a result, in C or C++, you never need to call `QueryInterface` to get to the base type if you do not want to, which can allow some performance improvements.
+
+> Note: WinRT interfaces do not follow this inheritance model. They are defined to follow the same model as the `[ComImport]`-based COM interop model in .NET.
+
+###### COM Interface Inheritance in .NET with <xref:System.Runtime.InteropServices.ComImportAttribute>
+
+In .NET, C# code that looks like interface inheritance isn't actually interface inheritance. Let's look at the following code:
+
+```csharp
+interface I
+{
+    void Method1();
+}
+interface J : I
+{
+    void Method2();
+}
+```
+
+This code does not say, "`J` implements `I`." The code actually says, "any type that implements `J` must also implement `I`." This difference leads to the fundamental design decision that makes interface inheritance in <xref:System.Runtime.InteropServices.ComImportAttribute>-based interop unergonomic. Interfaces are always considered on their own; an interface's base interface list has no impact on any calculations to determing a virtual function table for a given .NET interface.
+
+As a result, the natural equivalent of the above provided C++ COM interface example leads to a different virtual function table layout.
+
+C# code:
+
+```csharp
+[ComImport]
+[InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+interface IComInterface
+{
+    void Method();
+    void Method2();
+}
+
+[ComImport]
+[InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+interface IComInterface2 : IComInterface
+{
+    void Method3();
+}
+```
+
+Virtual Function Table layouts:
+
+| `IComInterface` Virtual Function Table slot | Method name |
+|-----------------------------|-------------|
+| 0 | `IUnknown::QueryInterface` |
+| 1 | `IUnknown::AddRef` |
+| 2 | `IUnknown::Release` |
+| 3 | `IComInterface::Method` |
+| 4 | `IComInterface::Method2` |
+
+| `IComInterface2` Virtual Function Table slot | Method name |
+|-----------------------------|-------------|
+| 0 | `IUnknown::QueryInterface` |
+| 1 | `IUnknown::AddRef` |
+| 2 | `IUnknown::Release` |
+| 3 | `IComInterface2::Method3` |
+
+As these virtual function tables differ from the C++ example, this will lead to serious problems at runtime. The correct definition of these interfaces in .NET with <xref:System.Runtime.InteropServices.ComImportAttribute> is as follows:
```

Comment:
```suggestion
As these virtual function tables differ from the C++ example, this will lead to serious problems at run time. The correct definition of these interfaces in .NET with <xref:System.Runtime.InteropServices.ComImportAttribute> is as follows:
```